### ISSUE TITLE
perf: replace repeated ones-stack pattern with single-allocation in all frameworks

### ISSUE THREAD

**Issue by hunter-heidenreich** [hunter-heidenreich/Kabsch-Cookbook#30]
## Goal

Replace the `torch.stack([ones] * (D-1) + [d_sign], dim=-1)` pattern (and its equivalents in JAX, TF, NumPy, MLX) with a single-allocation approach.

## Background

In every `kabsch` and `kabsch_umeyama` implementation, the reflection correction diagonal is built by stacking D-1 identical `ones` tensors with a sign vector:

```python
# pytorch/kabsch_svd_nd.py:178
B_diag = torch.stack([ones] * (D - 1) + [d_sign], dim=-1)  # BxD

# jax/kabsch_svd_nd.py:137
B_diag = jnp.stack([ones] * (D - 1) + [d_sign], axis=-1)
```

This creates D-1 redundant `ones` tensors in Python, each needing a separate allocation and a stack op. For D=3 it's only 2 allocations -- cheap in absolute terms -- but it's semantically wasteful and the pattern repeats across 10+ call sites in the codebase.

The cleaner approach is a single buffer fill + index assignment:

```python
# PyTorch equivalent
B_diag = torch.ones(B, D, dtype=P.dtype, device=P.device)
B_diag[:, -1] = d_sign
```

This is one allocation, one scalar write, no Python list construction.

## Scope

All occurrences across:
- `pytorch/kabsch_svd_nd.py` (2 occurrences)
- `jax/kabsch_svd_nd.py` (2 occurrences)
- `tensorflow/kabsch_svd_nd.py` (2 occurrences)
- `numpy/kabsch_svd_nd.py` (2 occurrences)
- `mlx/kabsch_svd_nd.py` -- different pattern already (uses `mx.concatenate`), assess separately

## Acceptance Criteria

- All occurrences replaced with the single-allocation pattern
- Existing tests pass unchanged (behavior is identical)
- No regressions in gradient tests (the `B_diag` participates in the autograd graph)
